src/analysis/processing/qgsalgorithmrandomraster.cpp

```diff
@@ -0,0 +1,313 @@
+      if ( ( qgsDoubleNear( mRandomLowerBound, 0.0 ) && qgsDoubleNear( mRandomUpperBound, 0.0 ) ) || qgsDoubleNear( mRandomUpperBound, mRandomLowerBound ) )
+      {
+        mRandomUpperBound = std::numeric_limits<float>::max();
+        mRandomLowerBound = std::numeric_limits<float>::min();
+      }
+      break;
+    case ( 6 ):
+      mRasterDataType = Qgis::Float64;
+      if ( ( qgsDoubleNear( mRandomLowerBound, 0.0 ) && qgsDoubleNear( mRandomUpperBound, 0.0 ) ) || qgsDoubleNear( mRandomUpperBound, mRandomLowerBound ) )
+      {
+        mRandomUpperBound = std::numeric_limits<double>::max();
+        mRandomLowerBound = std::numeric_limits<double>::min();
+      }
+      break;
+    default:
+      break;
+  }
+
+  mRandomIntDistribution = std::uniform_int_distribution<long>( mRandomLowerBound, mRandomUpperBound );
+  mRandomDoubleDistribution = std::uniform_real_distribution<double>( mRandomLowerBound, mRandomUpperBound );
+
+  return true;
+}
+
+
+QVariantMap QgsRandomRasterAlgorithm::processAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback *feedback )
+{
+  const QString outputFile = parameterAsOutputLayer( parameters, QStringLiteral( "OUTPUT" ), context );
+  QFileInfo fi( outputFile );
+  const QString outputFormat = QgsRasterFileWriter::driverForExtension( fi.suffix() );
+
+  int rows = std::max( std::ceil( mExtent.height() / mPixelSize ), 1.0 );
+  int cols = std::max( std::ceil( mExtent.width() / mPixelSize ), 1.0 );
+
+  //build new raster extent based on number of columns and cellsize
+  //this prevents output cellsize being calculated too small
+  QgsRectangle rasterExtent = QgsRectangle( mExtent.xMinimum(), mExtent.yMaximum() - ( rows * mPixelSize ), mExtent.xMinimum() + ( cols * mPixelSize ), mExtent.yMaximum() );
+
+  std::unique_ptr< QgsRasterFileWriter > writer = qgis::make_unique< QgsRasterFileWriter >( outputFile );
+  writer->setOutputProviderKey( QStringLiteral( "gdal" ) );
+  writer->setOutputFormat( outputFormat );
+  std::unique_ptr<QgsRasterDataProvider > provider( writer->createOneBandRaster( mRasterDataType, cols, rows, rasterExtent, mCrs ) );
+  if ( !provider )
+    throw QgsProcessingException( QObject::tr( "Could not create raster output: %1" ).arg( outputFile ) );
+  if ( !provider->isValid() )
+    throw QgsProcessingException( QObject::tr( "Could not create raster output %1: %2" ).arg( outputFile, provider->error().message( QgsErrorMessage::Text ) ) );
+
+  double step = rows > 0 ? 100.0 / rows : 1;
+
+  for ( int i = 0; i < rows ; i++ )
+  {
+    if ( feedback->isCanceled() )
+    {
+      break;
+    }
+    //prepare raw data depending on raster data type
+    QgsRasterBlock block( mRasterDataType, cols, 1 );
+    switch ( mTypeId )
+    {
+      case ( 0 ):
+      {
+        std::vector<quint8> byteRow( cols );
+        for ( int i = 0; i < cols; i++ )
+        {
+          byteRow[i] = static_cast<quint8>( mRandomIntDistribution( mRandomDevice ) );
```

Comment:
see above note about use of the c++ random classes

Reply:
fixed